docs/monitoring/faq/webhooks.mdx

```diff
@@ -77,3 +77,116 @@ When delivering events to your webhook endpoint we follow these guidelines
 - If your endpoint returns a 5xx status code in less than 5 seconds we retry up to 2 times with exponential backoff.
 - If your endpoint returns a 4xx status code, we declare the delivery failed and do not retry.
 - If your endpoint returns any body do not read it
+
+## Example with Modal
+
+### Setup
+
+For an example of how to set this up, we will use [Modal](https://modal.com/).
+
+First, create a Modal account. Then, locally install the Modal SDK:
+
+```shell
+pip install modal
+```
+
+### Secrets
+
+Next, you will need to set up some secrets.
+
+First, we will set up a secret to use to authenticate to our server.
+You can see instructions for secrets [here](https://modal.com/docs/guide/secrets).
+For this purpose, let's call our secret `ls-webhook` and have it set an environment variable with the name `LS_WEBHOOK`.
+
+We can also set up a LangSmith secret - luckily there is already an integration template for this!
+
+![LangSmith Modal Template](../static/modal_langsmith_secret.png)
+
+### Service
+
+After that, you can create a Python file that will serve as your endpoint.
+An example is below, with comments explaining what is going on:
+
+```python
```

Comment:
Cleaner version, without the need for an async wrapper, with langsmith dependency moved into Modal, and with a few unneeded imports removed.

```python
from fastapi import HTTPException, status, Request, Query
from modal import Secret, Stub, web_endpoint, Image

stub = Stub("auth-example", image=Image.debian_slim().pip_install("langsmith"))


@stub.function(
    secrets=[Secret.from_name("ls-webhook"), Secret.from_name("my-langsmith-secret")]
)
# We want this to be a `POST` endpoint since we will post data here
@web_endpoint(method="POST")
# We set up a `secret` query parameter
def f(runs: dict, secret: str = Query(...)):
    # You can import dependencies you don't have locally inside Modal funxtions
    from langsmith import Client

    # First, we validate the secret key we pass
    import os

    if secret != os.environ["LS_WEBHOOK"]:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Incorrect bearer token",
            headers={"WWW-Authenticate": "Bearer"},
        )

    # This is where we put the logic for what should happen inside this webhook
    ls_client = Client()
    ids = [r["id"] for r in runs]
    feedback = list(ls_client.list_feedback(run_ids=ids))
    for r, f in zip(runs, feedback):
        try:
            ls_client.create_example(
                inputs=r["inputs"],
                outputs={"output": f.correction},
                dataset_name="classifier-github-issues",
            )
        except Exception:
            raise ValueError(f"{r} and {f}")
    # Function body
    return "success!"
```

Reply:
The changes here look to be compatible with the full example in `classification.mdx`.